modules/apps/translation/translation-web/src/main/resources/META-INF/resources/asset/full_content.jsp

```diff
@@ -100,60 +100,64 @@ ViewTranslationDisplayContext viewTranslationDisplayContext = (ViewTranslationDi
 				boolean html = viewTranslationDisplayContext.getBooleanValue(infoField, TextInfoFieldType.HTML);
 				String label = viewTranslationDisplayContext.getInfoFieldLabel(infoField);
 				boolean multiline = viewTranslationDisplayContext.getBooleanValue(infoField, TextInfoFieldType.MULTILINE);
-			%>
 
-				<clay:row>
-					<clay:col
-						md="6"
-					>
+				String sourceContentDir = LanguageUtil.get(viewTranslationDisplayContext.getSourceLocale(), "lang.dir");
 
-						<%
-						String sourceContent = viewTranslationDisplayContext.getStringValue(infoField, viewTranslationDisplayContext.getSourceLocale());
-						String sourceContentDir = LanguageUtil.get(viewTranslationDisplayContext.getSourceLocale(), "lang.dir");
-						%>
-
-						<c:choose>
-							<c:when test="<%= html %>">
-								<label class="control-label">
-									<%= label %>
-								</label>
-
-								<div class="translation-editor-preview" dir="<%= sourceContentDir %>">
-									<%= sourceContent %>
-								</div>
-							</c:when>
-							<c:otherwise>
-								<aui:input dir="<%= sourceContentDir %>" label="<%= label %>" name="<%= label %>" readonly="true" tabIndex="-1" type='<%= multiline ? "textarea" : "text" %>' value="<%= sourceContent %>" />
-							</c:otherwise>
-						</c:choose>
-					</clay:col>
+				List<String> sourceStringValues = viewTranslationDisplayContext.getStringValues(infoField, viewTranslationDisplayContext.getSourceLocale());
 
-					<clay:col
-						md="6"
-					>
+				Iterator<String> sourceStringValuesIterator = sourceStringValues.iterator();
 
-						<%
-						String targetContent = viewTranslationDisplayContext.getStringValue(infoField, viewTranslationDisplayContext.getTargetLocale());
-						%>
-
-						<c:choose>
-							<c:when test="<%= html %>">
-								<label class="control-label">
-									<%= label %>
-								</label>
-
-								<div class="translation-editor-preview" dir="<%= LanguageUtil.get(viewTranslationDisplayContext.getTargetLocale(), "lang.dir") %>">
-									<%= targetContent %>
-								</div>
-							</c:when>
-							<c:otherwise>
-								<aui:input dir='<%= LanguageUtil.get(viewTranslationDisplayContext.getTargetLocale(), "lang.dir") %>' label="<%= label %>" name="<%= label %>" readonly="true" tabIndex="-1" type='<%= multiline ? "textarea" : "text" %>' value="<%= targetContent %>" />
-							</c:otherwise>
-						</c:choose>
-					</clay:col>
-				</clay:row>
+				List<String> targetStringValues = viewTranslationDisplayContext.getStringValues(infoField, viewTranslationDisplayContext.getTargetLocale());
+
+				Iterator<String> targetStringValuesIterator = targetStringValues.iterator();
+
+				while (sourceStringValuesIterator.hasNext() && targetStringValuesIterator.hasNext()) {
```

Comment:
maybe it's worth a try to make a single method that returns a structure with 2 fields (a array , map, etc..) on the display context

Reply:
Hi @AliciaGarciaGarcia I thought about it as well, however, it could be the case that the number of source and target values are not the same, and because of this we have to use separate iterators. Please see [this part of the code](https://github.com/liferay/liferay-portal/blob/master/modules/apps/translation/translation-web/src/main/resources/META-INF/resources/translate.jsp#L146-L162) from where I took the example.
Please let me know what you think.